iceoryx_examples/icedelivery/README.md

```diff
@@ -242,25 +227,62 @@ The typed publisher application is an example for a high-level user API and does
 described before. In this summary, just the differences to the prior publisher application are described.
 
 Starting again with the includes, there is now a different one:
+
 ```cpp
 #include "iceoryx_posh/popo/publisher.hpp"
 ```
 
 When it comes to the runtime, things are the same as in the untyped publisher. However, a typed publisher object is
-created
+created and the transmitted data type is provided as template parameter:
+
 ```cpp
 iox::popo::Publisher<RadarObject> publisher({"Radar", "FrontLeft", "Object"});
 ```
 
+The topic type must be default- and copy-constructible when the typed API is used.
+
 A similar while-loop is used to send the data to the subscriber. In contrast to the untyped publisher the typed one
-offers two additional possibilities
+offers three additional possibilities.
+
+#### #2 Functional approach with loaning
+
+Usage #2 constructs the data type with the values provided in loan:
+
+```cpp
+auto result = publisher.loan(ct, ct, ct);
+if (!result.has_error())
+{
+    auto& sample = result.value();
+    sample->x = ct;
+    sample->y = ct;
+    sample->z = ct;
+    sample.publish();
+}
+else
+{
+    // ..
+}
+```
+
+One might wonder what the type of the variable `sample` is? It is `iox::popo::Sample<RadarObject>`. This class behaves
+similar to a [`std::unique_ptr`](https://en.cppreference.com/w/cpp/memory/unique_ptr) and makes sure that the ownership
+handling is done automatically and memory is freed when going out of scope on subscriber side.
+
+#### #4 Publish by copy
```

Comment:
Where is `#1` and `#3`?

Reply:
These are already introduced in the untyped section, but I can add it again to make it clearer.

Reply:
Reworked by only using `and_then` / `or_else`